lib/gds_api/mapit.rb

```diff
@@ -5,10 +5,11 @@ class GdsApi::Mapit < GdsApi::Base
 
   def location_for_postcode(postcode)
     response = get_json("#{base_url}/postcode/#{CGI.escape postcode}.json")
-    return Location.new(response) unless response.nil?
+    Location.new(response) unless response.nil?
+
   rescue GdsApi::HTTPErrorResponse => e
     # allow 400 errors, as they can be invalid postcodes people have entered
-    raise e unless e.code == 400
+    e
```

Comment:
It's a little odd to have this method return the exception object instead of raising the exception. It would be more consistent with other adapters for this to simple raise pass the exception up (ie just get rid of the `rescue` clause entirely.